File: apps/api/src/modules/mcp/mcp.controller.ts
import { Body, Controller, Delete, Get, Param, Post } from '@nestjs/common';
import { McpService } from './mcp.service.js';
import type { McpTransportType } from './mcp.types.js';

@Controller('mcp/servers')
export class McpController {
  constructor(private readonly mcpService: McpService) {
    this.listServers = this.listServers.bind(this);
    this.getServer = this.getServer.bind(this);
    this.registerServer = this.registerServer.bind(this);
    this.discover = this.discover.bind(this);
    this.remove = this.remove.bind(this);
  }

  @Get()
  listServers() {
    return this.mcpService.listServers();
  }

  @Get(':id')
  getServer(@Param('id') id: string) {
    return this.mcpService.getServer(id);
  }

  @Post()
  registerServer(
    @Body()
    body: {
      name: string;
      code: string;
      transportType: McpTransportType;
      endpoint: string;
      authType?: string;
      metadata?: Record<string, unknown>;
    },
  ) {
    return this.mcpService.registerServer(body);
  }

  @Post(':id/discover')
  discover(@Param('id') id: string) {
    return this.mcpService.discoverCapabilities(id);
  }

  @Delete(':id')
  remove(@Param('id') id: string) {
    return this.mcpService.removeServer(id);
  }
}

File: apps/server-java/src/main/java/com/haoyitec/agent/server/module/mcp/McpController.java
package com.haoyitec.agent.server.module.mcp;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/v1/mcp/servers")
public class McpController {

    private final McpService mcpService;

    @GetMapping
    public Object listServers() {
        return mcpService.listServers();
    }

    @GetMapping("/{id}")
    public Object getServer(@PathVariable String id) {
        return mcpService.getServer(id);
    }

    @PostMapping
    public Object registerServer(@RequestBody Map<String, Object> body) {
        return mcpService.registerServer(body);
    }

    @PostMapping("/{id}/discover")
    public Object discover(@PathVariable String id) {
        return mcpService.discoverCapabilities(id);
    }

    @DeleteMapping("/{id}")
    public Object remove(@PathVariable String id) {
        return mcpService.removeServer(id);
    }
}

